larray/tests/test_session.py

```diff
@@ -134,24 +136,39 @@ def test_names(self):
 
     def test_h5_io(self):
         fpath = abspath('test_session.h5')
-
         self.session.save(fpath)
+
         s = Session()
         s.load(fpath)
         # HDF does *not* keep ordering (ie, keys are always sorted)
         self.assertEqual(list(s.keys()), ['e', 'f', 'g'])
 
+        # update an array (overwrite=False)
+        s['e'] = self.e2
+        s.save(fpath, overwrite=False)
+        s.load(fpath)
```

Comment:
This test (and all similar tests below) is a bit misleading. It will pass even if the file is completely wiped out, because the arrays still exist in the session s and s.load(path) does not clear the session. Maybe we want to change that last point, but that would be for another PR/release.